source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -9,34 +11,34 @@ AlternateProtocolsCacheImpl::AlternateProtocolsCacheImpl(TimeSource& time_source
 AlternateProtocolsCacheImpl::~AlternateProtocolsCacheImpl() = default;
 
 void AlternateProtocolsCacheImpl::setAlternatives(const Origin& origin,
-                                                  const std::vector<AlternateProtocol>& protocols,
-                                                  const MonotonicTime& expiration) {
-  Entry& entry = protocols_[origin];
-  if (entry.protocols_ != protocols) {
-    entry.protocols_ = protocols;
-  }
-  if (entry.expiration_ != expiration) {
-    entry.expiration_ = expiration;
-  }
+                                                  const std::vector<AlternateProtocol>& protocols) {
+  // TODO(RyanTheOptimist): Figure out the best place to enforce this limit.
```

Comment:
Unless I'm misreading the code, this ASSERT is reachable from untrusted input, which means that anyone can use it to crash an Envoy debug build? Can we instead:

```
  if (protocols.size() > 10) {
    ASSERT(false);
    protocols.resize(10);
  }
```

Reply:
This code is currently unreachable in Envoy so nobody will be crashing it :) 

But yes, as part of resolving alyssa's most recent comment, I'll make the cache truncate the list of protocols. (And I don't think we need an ASSERT at all)

Reply:
That sounds good, can you log a warning when truncating?